tests/autoscaler/autoscaler_test.go

```diff
@@ -0,0 +1,199 @@
+package ingress
+
+import (
+	"context"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+	"time"
+
+	"github.com/giantswarm/micrologger"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/sonobuoy-plugin/v5/tests/ctrlclient"
+)
+
+const (
+	helloWorldNamespace      = "default"
+	helloWorldDeploymentName = "helloworld"
+)
+
+func Test_Autoscaler(t *testing.T) {
+	var err error
+
+	ctx := context.Background()
+
+	tcCtrlClient, err := ctrlclient.CreateTCCtrlClient(ctx)
+	if err != nil {
+		t.Fatalf("error creating TC k8s client: %v", err)
+	}
+
+	logger, err := micrologger.New(micrologger.Config{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	logger.Debugf(ctx, "Creating %s deployment", helloWorldDeploymentName)
```

Comment:
Lately I've been thinking that maybe we should use the test logger instead of our own to get a consistent experience when running `go test -v`, like here https://github.com/giantswarm/sonobuoy-plugin/blob/master/tests/cptcconnectivity/cptcconnectivity_test.go#L53

Reply:
any opinions on this?

Reply:
could you expand what "consistent experience" means in this case?

Reply:
Would it make sense to create a wrapper in `micrologger` that captures `t` and uses `t.Log()` underneath? Then the code would be consistent while the output behavior would also be consistent in different use cases?

Reply:
I still don't get what's the difference between t.logger and micrologger's logger :(

Reply:
I believe at least the effects of verbose vs. non-verbose test output:
```
$ go help testflag | grep -A3 -- '-v'|head -n3
	-v
	    Verbose output: log all tests as they are run. Also print all
	    text from Log and Logf calls even if the test succeeds.
```

...but there's also the difference between formatting.

Reply:
Yeah. Tests will only output logs when enabling verbose execution, but using our own logger changes that semantic. Also, not using our logger means not using our log format for tests, not sure if really important. I think it'd be better to use the test logger, but it's not important.